### DISCUSSION TITLE
Welcome to academictwitteR Discussions!

### DISCUSSION THREAD

**Discussion by cjbarrie** [cjbarrie/academictwitteR#45] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

In short, please use this as an alternative space to opening an Issue: you can suggest things to which you don't have a solution or might not be an Issue, and the Community can generally interact to help make `academictwitteR` better :)

**Comment by tyskevdb** [comment]
Dear cjbarrie,

Thanks for this gem of a package! Really appreciated. 

I was just wondering how I could use academictwitteR to collect replies that all use the same conversation_id (e.g. replies to one influential Tweet). https://developer.twitter.com/en/docs/twitter-api/conversation-id 

Thank you,
Tijs

**Reply by cjbarrie** [reply]
Hey! Thanks for this. Looking into this as we speak. There have been some reports of issues on this query from the client side so we'll have to work this out.

**Reply by cjbarrie** [reply]
An update on this: currently searching by conversation ID remains highly restrictive and is only available in the v2 endpoint for last 7 days. This has been noted as highly restrictive by other users (see here: https://twittercommunity.com/t/unable-to-scrape-filtered-stream-using-conversation-id/146971/2) and we'll continue to monitor whether anything changes

**Comment by laurebokobza** [comment]
Hi Chris & Justin, thanks a lot for the package, I've used it this week and it's absolutely brilliant!

I was wondering whether there is any way to get the (approximate) location from the "place_id" variable for those tweets gathered with academicTwitteR that don't have a precise location pin (long/lat), but do come with this geographic information? Something equivalent to https://developer.twitter.com/en/docs/twitter-api/v1/geo/place-information/api-reference/get-geo-id-place_id  - but for R?

Cheers,
Laure

**Reply by cjbarrie** [reply]
Hey! This is something we need to work on. The good news is the data is there but the name of the place is returned in the users_ information (only stored when using a "data_path" argument). If you bind up the users_* jsons you will see that there is a location listed for given users. 

So, something like:

```
tweets <- get_geo_tweets("protest", "2020-01-01T00:00:00Z", 
                         "2020-01-05T00:00:00Z", bearer_token, 
                         data_path = "data/", bind_tweets = F)

utweets <- bind_user_jsons("data/")

tweets <- bind_tweet_jsons("data/")

tids <- unique(tweets$author_id)

uids <- unique(utweets$id)

utweetsf <- utweets %>%
  filter(id %in% tids) %>%
  rename("author_id" = id) %>%
  select(author_id, location) %>%
  distinct()

tweetsm <- merge(tweets, utweetsf, by = "author_id")
```

will combine the locations with the geo place_id. This is less than ideal so we'll have to think of a way to bind this information more cleanly.

**Reply by laurebokobza** [reply]
Thanks a lot for the reply!! And amazing work on the package

**Comment by AfonsoBiscaia** [comment]
Hi,

Thank you for developing this package. I'm sure it will be very helpful for my research, even though I'm still very much a beginner at R!

However, when I was checking out the package's functionalities, I noticed that when I run get_user_tweets, it then places the variables at random on the table, which isn't practical if you want to build a spreadsheet based on a series of collections. Is there any way to force the package to always use the same columns for the same variables, or am I just not doing something I should have done?

Thanks in advance,
Afonso

**Reply by cjbarrie** [reply]
Hi Afonso,

Thanks for this. We noticed that columns are ordered differently for separate queries and will add this to our things to amend. Column names will be consistent throughout, however, despite being differently ordered so you will still be able to bind them with relative ease. 

As a MWE:

```
# sample data, variable c is in df1, variable d is in df2, and the data.frames are differently ordered on b
df1 = data.frame(b=6:10, a=1:5, d=month.name[1:5])
df2 = data.frame(a=6:10, b=16:20, c = letters[8:12])

df <- bind_rows(df1,df2)

```

**Reply by AfonsoBiscaia** [reply]
Thank you for the quick and thoughtful answer, CJ!

**Comment by lightsociologist** [comment]
Hi!

Are there plans to include capturing users followers and followed lists? Or is this already present and I'm missing it? I've been digging into the package for the past couple of hours and am really impressed!

Thanks,

Ryan

**Comment by fpbasi** [comment]
Hello @cjbarrie. How can I determine the username in dataframe? I see that author_id and conversation_id in the frame, however, i do not know, how can I find the Twitter username for the tweet?

**Reply by cjbarrie** [reply]
Hi @fpbasi the usernames are delivered as part of the user_* payload (when outputting to JSON). This is a client-side problem that we can't really solve very easily at present and are monitoring if Twitter changes how API delivers the data. 

As a workaround: you can take the author_id and merge with author_id and username in the user_* JSON data.

**Comment by jcs82** [comment]
Hi, thanks for the package - works really well. I just have one issue - I am searching for 'has:geo' tweets, so only ones with place_id information. The results have all the geo data including the place_ids. However, the results are missing the "place.fields" ---- ie: ("contained_within,country,country_code,full_name,geo,id,name,place_type"). It looks these feilds are requested in the "get_all_tweets" function, however... Any ideas?! Thanks again, Jon

**Comment by shmuhammadd** [comment]
Hi Thanks for the package and the effort to improve it. 

I want search for tweets containing emoticons ( '-:)' ) and I find that it is not working.  It gives me an error:  `Status code: 400`

Please, can you kindly advise. The code is down below :

tweet_emoticon <- get_all_tweets( 
'-:)',
is_retweet = FALSE, 
"2019-06-1T00:00:00Z",
"2021-06-12T00:00:00Z",
data_path = "data/",
n = 100
  )

Best,
Shamsuddeen
